lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
```

Comment:
@WeatherGod Sure, take a look at the second comment (https://github.com/matplotlib/matplotlib/pull/4118#issuecomment-74774247).  This gives cases for testing.  The (main) bug appears on the line where I write ``# Should *also* print 1`` as it does not, it gives a different cid to the first, i.e. 2.